syntax = "proto3";
option java_multiple_files = true;
option objc_class_prefix = "HLW";

package streamServer;

service Datastreamer {
    rpc GetData (DataRequest) returns (stream DataResponse) {}
    rpc UpdateRealTimeAnalytics (RealTimeAnalyticsRequest) returns (Confirmation) {}
    rpc UpdateRollingAnalytics (RollingAnalyticsRequest) returns (Confirmation) {}
    rpc GetAnalytics (DataRequest) returns (AnalyticsResponse) {}
}

message DataRequest{
    string name = 1;
}

message AnalyticsResponse{
    string analytics = 1;
}

message RollingAnalyticsRequest{
    string avgWordLength = 1;
}

message Confirmation{
    string confirm = 1;
}

message RealTimeAnalyticsRequest{
    string avgWordsPerPost = 1;
    string postWithMostWords = 2;
    string authorWithMostDeletedPosts = 3;
}

message DataResponse{
    string id = 1;
    string title = 2;
    string score = 3;
    string author = 4;
    string author_flair_text = 5;
    string removed_by = 6;
    string total_awards_received = 7;
    string awarders = 8;
    string created_utc = 9;
    string full_link = 10;
    string num_comments = 11;
    string over_18 = 12;
}